### ISSUE TITLE
aft_codebuild_customizations_policy is invalid and causes permission issues

### ISSUE THREAD

**Issue by engrun** [aws-ia/terraform-aws-control_tower_account_factory#137]
**AFT Version: 1.3.5**

**Terraform Version & Provider Versions**
Please provide the outputs of `terraform version` and `terraform providers` from within your AFT environment

`terraform version`
```
Terraform v1.1.8
on darwin_arm64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.75.1
+ provider registry.terraform.io/hashicorp/local v2.2.2
+ provider registry.terraform.io/hashicorp/random v3.1.2
+ provider registry.terraform.io/hashicorp/time v0.7.2

```

`terraform providers`
```

Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── provider[registry.terraform.io/hashicorp/local]
├── module.aft_ssm_parameters
│   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── module.aft_account_provisioning_framework
│   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── module.aft_backend
│   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── module.aft_iam_roles
│   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
│   ├── module.log_archive_exec_role
│       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
│   ├── module.aft_exec_role
│       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
│   ├── module.audit_exec_role
│       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
│   └── module.ct_management_exec_role
│       └── provider[registry.terraform.io/hashicorp/aws] >= 2.7.0
├── module.aft_feature_options
│   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── module.aft_lambda_layer
│   ├── provider[registry.terraform.io/hashicorp/random]
│   ├── provider[registry.terraform.io/hashicorp/local]
│   └── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
├── module.packaging
│   └── provider[registry.terraform.io/hashicorp/archive]
├── module.aft_code_repositories
│   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
│   └── provider[registry.terraform.io/hashicorp/local]
├── module.aft_account_request_framework
│   ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
│   └── provider[registry.terraform.io/hashicorp/time]
└── module.aft_customizations
    ├── provider[registry.terraform.io/hashicorp/aws] >= 3.72.0, < 4.0.0
    └── provider[registry.terraform.io/hashicorp/local]

Providers required by state:

    provider[registry.terraform.io/hashicorp/local]

    provider[registry.terraform.io/hashicorp/random]

    provider[registry.terraform.io/hashicorp/archive]

    provider[registry.terraform.io/hashicorp/time]

    provider[registry.terraform.io/hashicorp/aws]


```

**Bug Description**

In file
`modules/aft-customizations/iam/role-policies/aft_codebuild_customizations_policy.tpl`

line 105, which looks like this: 
`  "arn:aws:dynamodb:${data_aws_caller_identity_current_account_id}:${data_aws_caller_identity_current_account_id}:table/aft*"`

Problem: 
Here "data_aws_caller_identity_current_account_id" is used twice, causing the IAM policy to be invalid. 
Instead the correct region should be injected, like this. 
`
"arn:aws:dynamodb:${data_aws_region_current_name}:${data_aws_caller_identity_current_account_id}:table/aft*"
`

*This causes an error in customizations scripts when trying to read the account request from the `aft-request` DynamoDB table.*

**To Reproduce**
1. Setup a pre-api-helpers.sh script
2. Make sure to assume creds for the AFT-management account before calling the script, like this, `./aws-aft-core-framework/sources/scripts/creds.sh --aft-mgmt`
3. In the script, execute `aws dynamodb get-item --table-name aft-request --key "{\"id\": {\"S\": \"$ACCOUNTEMAIL\"}}"`.
4. Run the optimization Step Function to trigger the pipeline. 
5. The command fails.

**Expected behavior**
I would expect the script to be able the fetch account-request data from the DynamoDB table.

**Comment by snebhu3** [comment]
@engrun thank you for reporting the bug. I have created an internal backlog to fix the malformed policy statement.

**Comment by stumins** [comment]
This issue has been resolved in the latest release, [1.3.6](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/releases/tag/1.3.6)
